blackbox/test_conditional.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
```

Comment:
We can do that, I just thought that it would make sense to have one general test for all things conditional.  What about leaving it as one script with separate functions?